asserts/database.go

```diff
@@ -89,6 +89,13 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// InvalidRevisionError describes an error related to the revision of an assertion.
+type InvalidRevisionError string
```

Comment:
ConflictingRevisionError? SupersededRevisionError?

given that we are introducing this, I would really make struct carrying the two relevant revno numbers (that we put in the error message)